Fix the test for population.CalculateRecruitment()

Bai-Li-NOAA opened the pull request:
<!---
Thanks for opening a pull request (PR) to FIMS!
Everything inside of the less than!--- and --greater than is an html comment to
help you navigate submitting this PR. This commented text as well as other
comments will **NOT** appear in the final PR. If you do not believe me, just
toggle between Write and Preview to see what your PR will look like without the
comments.

General instructions are as follows:
* Please read the html comment under each heading and follow the instructions.
* For smaller changes, feel free to skip sections not flagged as "MANDATORY".
* Before opening the PR, make sure all the GitHub actions are passing on the
  remote feature branch.
* Make sure this PR has an informative title rather than the default text.
-->

# What is the feature?
<!--- MANDATORY -->
<!---
Please briefly describe the feature using bullet points.
-->
This PR addresses issue #494 and fixes the test for population.CalculateRecruitment().

# How have you implemented the solution?
<!---
Provide relevant information for each file that has changed.
-->

- update the C++ test [here](https://github.com/NOAA-FIMS/FIMS/blob/3dbecd32218697b992877f2395b6ada8d3587588/tests/gtest/test_population_Recruitment.cpp)
- register the test to [CMakeLists.txt](https://github.com/NOAA-FIMS/FIMS/blob/3dbecd32218697b992877f2395b6ada8d3587588/tests/gtest/CMakeLists.txt#L246C1-L256C45) to execute the test when running GoogleTest

# Does the PR impact any other area of the project?
<!---
Describe how features of FIMS are impacted by this change.
Commonly, changes will impact inputs or outputs to or from a FIMS model.
You can use subheadings to help clearly articulate which area of FIMS your PR
impacts. Feel free to use the example below to get started describing your
changes.

## Input
Text describing changes to the input.

## Output
Text describing changes to the output.
-->
No.

# How to test this change
<!--
Please include a test file and/or GitHub workflow. Files can be zipped and
uploaded directly to the PR.
-->
The C++ test can be found [here](https://github.com/NOAA-FIMS/FIMS/blob/3dbecd32218697b992877f2395b6ada8d3587588/tests/gtest/test_population_Recruitment.cpp).
[All GHA checks](https://github.com/NOAA-FIMS/FIMS/actions?query=branch%3A494-questionhelp-why-is-populationspawning_biomass-indexed-by-age-in-test) have passed. 

# Developer pre-PR checklist
<!-- 
Please do these steps locally for big changes.
For more details see
https://noaa-fims.github.io/collaborative_workflow/contributor-guidelines.html#code-development
Note that GitHub Actions does all of these checks when pushing to FIMS.
If you do any of the following, uncomment each relative line to acknowledge that you did it.
-->
- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.
<!-- - [x] Ran [cmake build and ctest locally](https://noaa-fims.github.io/collaborativ/e_workflow/testing.html#c-unit-testing-and-benchmarking) and the C++ tests passed -->
<!-- - [x] Ran `devtools::document()` locally and pushed changes to this remote feature branch -->
<!-- - [x] Ran `styler::style_pkg()` locally, committed the changes in a separate commit, and pushed the commit to this remote feature branch. -->
<!-- - [x] Ran `devtools::check()` locally and the package compiled and all R tests passed. If there are failing tests, run `devtools::test(filter = "file_name")` (where "test-file_name.R" is the file containing the failing tests) to troubleshoot tests. -->

iantaylor-NOAA commented on inst/include/population_dynamics/population/population.hpp:
```diff
@@ -377,7 +377,7 @@ struct Population : public fims_model_object::FIMSObject<Type> {
     FIMS_LOG << "recruitment 2" << std::endl;
     FIMS_LOG << "phi0 = " << phi0 << std::endl;
-    FIMS_LOG << "spawning biomass = " << this->spawning_biomass[year]
+    FIMS_LOG << "spawning biomass = " << this->spawning_biomass[year - 1]
```
This change seems to match what's noted in https://noaa-fims.github.io/collaborative_workflow/m1-model-specification.html#beverton-holt-recruitment-function, but I might change it to 
"spawning_biomass[year - 1] = " to make it extra clear what's being logged.

Bai-Li-NOAA replied:
Agree. The text has been changed to "spawning_biomass[year - 1] = ".

iantaylor-NOAA commented on tests/gtest/test_population_Recruitment.cpp:
```diff
@@ -8,70 +8,49 @@ namespace
+        int r_year = 5;
+        int r_age = 0; // just calculating for age 0 
+        int r_i_age_year = r_year * population.nages + r_age;
+        int r_i_agem1_yearm1 = (r_year - 1) * population.nages + r_age - 1;
```
Is this line needed? I don't see `r_i_agem1_yearm1` being used anywhere else in the file.

Bai-Li-NOAA replied:
No, `r_i_agem1_yearm1` is not needed. I have removed this line of code. Thanks for catching it.

iantaylor-NOAA commented on tests/gtest/test_population_Recruitment.cpp:
```diff
@@ -8,70 +8,49 @@ namespace
-        int year = 4;
-        int age = 0; // just calculating for age 0 
+        // calculating spawning biomass for year 4
+        int sb_year = 4;
```
Separating `sb_year = 4`  from `r_year = 5` makes sense to keep things clear.

iantaylor-NOAA reviewed (approved):
Changes look good. Thanks @Bai-Li-NOAA for sorting all this so quickly.

iantaylor-NOAA commented:
Note: based on conversation during yesterday's meeting, now that I've completed the review, I've unassigned myself and assigned @Bai-Li-NOAA so she can choose how/whether to address the minor comments in the unresolved conversations. If I didn't have any comments, I would have just merged since I was the assignee. Once @Bai-Li-NOAA has resolved the conversations, she could either merge the PR or re-assign me if further input is needed. 

Please let me know if that workflow doesn't match how we're thinking of using PR assignments.

Bai-Li-NOAA commented:
Thank you, @iantaylor-NOAA, for your review. I've incorporated your suggestions and updated the code. Additionally, I've resolved the merge conflicts. The workflow makes sense to me and I have merged the changes into the main branch.
